api4/system_test.go

```diff
@@ -25,10 +25,9 @@ func TestGetPing(t *testing.T) {
 	th := Setup(t)
 	defer th.TearDown()
 
-	t.Run("basic ping", func(t *testing.T) {
-
+	th.TestForAllClients(t, func(t *testing.T, client *model.Client4) {
 		t.Run("healthy", func(t *testing.T) {
-			status, resp := th.Client.GetPing()
+			status, resp := client.GetPing()
```

Comment:
What do you think of using `th.TestForAllClients` instead of using `t.Run`s for each of them. We can save some indentation here :)

Reply:
We can make that change, but I kinda like how easy to grasp and grouped the test output is with this schema:

```
--- PASS: TestGetPing (0.85s)
    --- PASS: TestGetPing/basic_ping/Client (0.03s)
        --- PASS: TestGetPing/basic_ping/Client/healthy (0.00s)
        --- PASS: TestGetPing/basic_ping/Client/unhealthy (0.03s)
    --- PASS: TestGetPing/basic_ping/SystemAdminClient (0.03s)
        --- PASS: TestGetPing/basic_ping/SystemAdminClient/healthy (0.00s)
        --- PASS: TestGetPing/basic_ping/SystemAdminClient/unhealthy (0.03s)
    --- PASS: TestGetPing/basic_ping/LocalClient (0.03s)
        --- PASS: TestGetPing/basic_ping/LocalClient/healthy (0.00s)
        --- PASS: TestGetPing/basic_ping/LocalClient/unhealthy (0.03s)
    --- PASS: TestGetPing/with_server_status/Client (0.00s)
        --- PASS: TestGetPing/with_server_status/Client/healthy (0.00s)
        --- PASS: TestGetPing/with_server_status/Client/unhealthy (0.00s)
    --- PASS: TestGetPing/with_server_status/SystemAdminClient (0.00s)
        --- PASS: TestGetPing/with_server_status/SystemAdminClient/healthy (0.00s)
        --- PASS: TestGetPing/with_server_status/SystemAdminClient/unhealthy (0.00s)
    --- PASS: TestGetPing/with_server_status/LocalClient (0.00s)
        --- PASS: TestGetPing/with_server_status/LocalClient/healthy (0.00s)
        --- PASS: TestGetPing/with_server_status/LocalClient/unhealthy (0.00s)
    --- PASS: TestGetPing/ping_feature_flag_test/Client (0.04s)
    --- PASS: TestGetPing/ping_feature_flag_test/SystemAdminClient (0.05s)
    --- PASS: TestGetPing/ping_feature_flag_test/LocalClient (0.03s)
```

No strong opinion here tho, so if you think we should split the cases, I'm happy to :slightly_smiling_face:

Reply:
This also makes sense, let's keep it as it is 👍